micrometer-tracing/src/main/java/io/micrometer/tracing/docs/DocumentedSpan.java

```diff
@@ -59,58 +68,27 @@
     }
 
     /**
-     * Returns required prefix to be there for events and tags. Example {@code foo.} would
-     * require the tags and events to have a {code foo} prefix like this for tags:
-     * {@code foo.bar=true} and {@code foo.started} for events.
-     * @return required prefix
+     * @return allowed tag keys - if set will override any tag keys coming from {@link DocumentedSpan#overridesDefaultSpanFrom()}
      */
-    default String prefix() {
-        return "";
-    }
-
-    /**
-     * Asserts on tags, names and allowed events.
-     * @param span to wrap
-     * @return wrapped span
-     */
-    default AssertingSpan wrap(Span span) {
-        if (span == null) {
-            return null;
-        }
-        else if (span instanceof AssertingSpan) {
-            return (AssertingSpan) span;
-        }
-        return AssertingSpan.of(this, span);
+    default TagKey[] getTagKeys() {
+        return new TagKey[0];
```

Comment:
Since an empty array is immutable, can we have a static final instance and return it instead of always creating a new one?